apps/web/src/features/courses/pages/components/course-description-modal.tsx

```diff
@@ -1,222 +1,609 @@
-import { BiCalendar } from 'react-icons/bi';
-import { BsClock, BsHourglass } from 'react-icons/bs';
-import { FaChalkboardTeacher } from 'react-icons/fa';
-import { HiOutlineBookOpen } from 'react-icons/hi';
-import { IoPricetagOutline } from 'react-icons/io5';
-
-import graduateImg from '../../../../assets/birrete.png';
-import watch from '../../../../assets/cloclk.png';
-import Book from '../../../../assets/courses/livre.svg?react';
-import rabitPen from '../../../../assets/rabbit_holding_pen.svg';
-import { Button } from '../../../../atoms/Button/index.tsx';
+import { QRCodeSVG } from 'qrcode.react';
+import {
+  type CSSProperties,
+  Children,
+  Fragment,
+  type PropsWithChildren,
+  useCallback,
+  useState,
+} from 'react';
+import { AiOutlineCopy } from 'react-icons/ai';
+
+import leftBackgroundImg from '../../../../assets/courses/left-background.png';
+import PlanBLogo from '../../../../assets/planb_logo_horizontal_black.svg?react';
 import { Modal } from '../../../../atoms/Modal/index.tsx';
 import { ReactPlayer } from '../../../../components/ReactPlayer/index.tsx';
 import { addSpaceToCourseId } from '../../../../utils/courses.ts';
-import { computeAssetCdnUrl } from '../../../../utils/index.ts';
+import { formatTime } from '../../../../utils/date.ts';
+import { computeAssetCdnUrl, trpc } from '../../../../utils/index.ts';
 import type { TRPCRouterOutput } from '../../../../utils/trpc.ts';
 
-type Course = NonNullable<TRPCRouterOutput['content']['getCourse']>;
-const { useGreater } = BreakPointHooks(breakpointsTailwind);
+const hexToBase64 = (hexstring: string) => {
+  return Buffer.from(hexstring, 'hex').toString('base64');
+};
+
+const getFormattedUnit = (amount: number, unit: string, floating = 2) => {
+  let prefix = '';
+  if (amount > 0 && amount < 0.01) {
+    amount = 0.01;
+    prefix = `< `;
+  }
+
+  if (unit === 'sats') {
+    return `${prefix}${amount} sats`;
+  }
+
+  return `${prefix}${Intl.NumberFormat(undefined, {
+    style: 'currency',
+    currency: unit,
+    currencyDisplay: 'narrowSymbol',
+    minimumFractionDigits: floating,
+    maximumFractionDigits: floating,
+  }).format(amount)}`;
+};
+
+const DEFAULT_CURRENCY = 'EUR';
+
+interface PaymentData {
+  id: string;
+  pr: string;
+  onChainAddr: string;
+  amount: number;
+  checkoutUrl: string;
+}
+
+interface WebSocketMessage {
+  settled: boolean;
+}
+
 interface CourseDescriptionModalProps {
   course: NonNullable<TRPCRouterOutput['content']['getCourse']>;
   satsPrice: number;
   isOpen: boolean;
   onClose: (isPaid?: boolean) => void;
-  onPay: (
-    satsPrice: number,
-    withPhysical: boolean,
-    part?: number,
-    chapter?: number,
-  ) => void;
 }
 
+const sideStyle: CSSProperties = {
+  position: 'absolute',
+  top: 0,
+  left: 0,
+  width: '50%',
+  height: '100%',
+};
+
+const borderStyle: CSSProperties = {
+  borderWidth: 1,
+  borderColor: 'rgba(255, 255, 255, 0.25)',
+  overflow: 'hidden',
+};
+
+const baseButtonStyle: CSSProperties = {
```

Comment:
We should use our "Button" component everywhere. I am also discussing with designer to have a proper design system and not to implement different button style everywhere.

Reply:
Done